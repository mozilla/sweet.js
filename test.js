import { id } from "./macros/id_mod.js";
import { * } from "./macros/stxcase.js";
import { map } from "./macros/helper.js" for macros;

macro m {
    case {_ } => {
        map([1, 2, 3], function(x) { return x; });
        return #{42};
    }
}
m

var x = id 24;
