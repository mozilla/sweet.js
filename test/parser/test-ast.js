import test from 'ava';
import { getAst } from '../assertions';

test('does not include the lang directive in the AST', t => {
  t.snapshot(
    getAst(`
    'lang sweet.js';
  `)
  );
});

test('does include the use strict directive in the AST', t => {
  t.snapshot(
    getAst(`
    'use strict';
  `)
  );
});
