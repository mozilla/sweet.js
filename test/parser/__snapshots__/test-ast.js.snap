// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does include the use strict directive in the AST 1`] = `
Module {
  "directives": Array [
    Object {
      "rawValue": "use strict",
      "type": "Directive",
    },
  ],
  "items": Array [],
  "loc": null,
  "type": "Module",
}
`;

exports[`does not include the lang directive in the AST 1`] = `
Module {
  "directives": Array [],
  "items": Array [],
  "loc": null,
  "type": "Module",
}
`;

exports[`includes export declaration in AST 1`] = `
Module {
  "directives": Array [],
  "items": Array [
    Export {
      "declaration": VariableDeclaration {
        "declarators": Array [
          VariableDeclarator {
            "binding": BindingIdentifier {
              "loc": null,
              "name": "x_0",
              "type": "BindingIdentifier",
            },
            "init": LiteralNumericExpression {
              "loc": null,
              "type": "LiteralNumericExpression",
              "value": 1,
            },
            "loc": null,
            "type": "VariableDeclarator",
          },
        ],
        "kind": "var",
        "loc": null,
        "type": "VariableDeclaration",
      },
      "loc": null,
      "type": "Export",
    },
  ],
  "loc": null,
  "type": "Module",
}
`;

exports[`includes export in AST 1`] = `
Module {
  "directives": Array [],
  "items": Array [
    ExportFrom {
      "loc": null,
      "moduleSpecifier": null,
      "namedExports": Array [
        ExportSpecifier {
          "exportedName": "b",
          "loc": null,
          "name": null,
          "type": "ExportSpecifier",
        },
      ],
      "type": "ExportFrom",
    },
  ],
  "loc": null,
  "type": "Module",
}
`;
